### PR TITLE
do not create qa cache directory until the cache is complete

### PR DESCRIPTION
**Pull request by gandrewstone**
... and fix return check on compare_exchange_weak

### CODE REVIEW

File: qa/rpc-tests/test_framework/util.py [hunk 1]
```diff
@@ -355,33 +355,46 @@ def initialize_chain(test_dir,bitcoinConfDict=None,wallets=None, bins=None):
...
+            if os.path.isdir(os.path.join("building_cache","node"+str(i))):
+                shutil.rmtree(os.path.join("building_cache","node"+str(i)))
 
         # Create cache directories, run bitcoinds:
+        logging.info("  starting 4 bitcoind")
         for i in range(4):
             for retry in range(4):
                 try:
-                    datadir=initialize_datadir("cache", i, bitcoinConfDict, wallets, bins)
+                    datadir=initialize_datadir("building_cache", i, bitcoinConfDict, wallets, bins)
```
Comment (ptschip) [inline qa/rpc-tests/test_framework/util.py:392]: I don't get the change to "building_cache" ? was "cache" giving a problem?

File: src/blockrelay/graphene.cpp [hunk 2]
```diff
@@ -1453,7 +1453,7 @@ bool HandleGrapheneBlockRequest(CDataStream &vRecv, CNode *pfrom, const CChainPa
             pfrom->nGetGrapheneLastTime = GetTime();
         uint64_t nNow = GetTime();
         double tmp = pfrom->nGetGrapheneCount;
-        while (pfrom->nGetGrapheneCount.compare_exchange_weak(
+        while (!pfrom->nGetGrapheneCount.compare_exchange_weak(
```
Comment (ptschip) [inline src/blockrelay/graphene.cpp:1456]: uggh, yeah I messed that up...thanks!

File: qa/rpc-tests/test_framework/util.py (hunk 1 above)
Comment (gandrewstone) [reply qa/rpc-tests/test_framework/util.py:392]: Yes, if cache construction was aborted then all subsequent runs would use the bad cache.  The result is weird failures of tests that are intolerant of a < 200 block cache.

**Review by ptschip — APPROVED** [review]
ah , i didn't see that you renamed building_cache to cache later.  looks good.

**Pull request merged**